include/oneapi/dpl/pstl/pstl_config.h

```diff
@@ -105,13 +106,19 @@
 #define _PSTL_EARLYEXIT_PRESENT (__INTEL_COMPILER >= 1800)
 #define _PSTL_MONOTONIC_PRESENT (__INTEL_COMPILER >= 1800)
 
-#if (__INTEL_COMPILER >= 1900 || !defined(__INTEL_COMPILER) && _PSTL_GCC_VERSION >= 40900 || _OPENMP >= 201307)
+#if (__INTEL_LLVM_COMPILER || __INTEL_COMPILER >= 1900 ||                                                              \
+     !defined(__INTEL_LLVM_COMPILER) && !defined(__INTEL_COMPILER) && _ONEDPL_GCC_VERSION >= 40900 ||                  \
+     _OPENMP >= 201307)
 #    define _PSTL_UDR_PRESENT 1
 #else
 #    define _PSTL_UDR_PRESENT 0
 #endif
 
-#define _PSTL_UDS_PRESENT (__INTEL_COMPILER >= 1900 && __INTEL_COMPILER_BUILD_DATE >= 20180626)
+#if (__INTEL_LLVM_COMPILER >= 20230100) || (__INTEL_COMPILER >= 1900 && __INTEL_COMPILER_BUILD_DATE >= 20180626)
+#    define _ONEDPL_UDS_PRESENT 1
+#else
+#    define _ONEDPL_UDS_PRESENT 0
+#endif
```

Comment:
Interestingly with 2023.1 on Windows (because that is where I have that version installed) and tbb backend, this results in a failure
```
.\build\test\scan.pass.exe
error at C:\Users\dhoeflin\source\repos\oneDPL\test\parallel_api\numeric\numeric.ops\scan.pass.cpp:137 - wrong result from inclusive_scan with binary operator, at index 1
```
If I disable UDS, it succeeds.  Its unclear if this is a windows issue or a 2023.1-> 2023.2 support issue.  Sadly I don't have 2023.1 on Linux, or 2023.2 on Windows to easily check.

Reply:
Looks likely to be a Windows issue, 2023.2 TBB backend also fails with the same error, and passes when UDS is turned off.  Still remains to check Linux 2023.1.

Reply:
I just checked on 2023.1 and 2023.2 on Linux, and it seems to succeed.
My guess is that this is a bug on Windows, but that 2023.1 is the appropriate version to enable UDS on Linux.

Reply:
Thanks, Dan. I have turned off UDS on Windows. Checking that we do not break anything with 2023.2 on Linux was also useful.